### DISCUSSION TITLE
how to set mitmweb to have an explicit https proxy

### DISCUSSION THREAD

**Discussion by zer0uno** [mitmproxy/mitmproxy#5947] [Q&A] [answered]
I've mitmweb running on localhost:8080 and I have correctly configured the OS (Ubuntu 20.04) to use it:
![2023-02-23_21-35](https://user-images.githubusercontent.com/15458756/221024532-14ddf41a-38b5-4a60-9072-e21dc7715f70.png).

Now when I make an HTTP request everything works like a sharm, in the browser and in mitmweb I can correctly see the request.
However, when I navigate to a site that uses HTTPS things start to not work anymore, that is I cannot see the request in mitmweb and in the browser I get the error messaging saying "it was not possible to set a secure communication".

So, is it possible to set mitmweb as an explicit https proxy? I'm not interested in seeing the HTTPS requests in mitmweb but I just want to proxy the HTTPS requests through it. As far as I understood in Explicit HTTPS Proxy it is not necessary to install a certificate on the client because:

> The proxy here is just a facilitator - it blindly forwards data in both directions without knowing anything about the contents. 

In addition from Wireshark I can see the HTTP Connect request:
![2023-02-23_22-01](https://user-images.githubusercontent.com/15458756/221029339-b507e035-2ac4-4c0d-8057-6f880bed130d.png)

**Comment by mhils** [comment, marked as answer]
> As far as I understood in Explicit HTTPS Proxy it is not necessary to install a certificate on the client because:

That depends. If you want the proxy to just pass through data without looking into it (a regular explicit proxy), this is true. But you can't decrypt connection contents. If you want to intercept the connection (like mitmproxy does by default), the client needs to trust mitmproxy's CA certificate.

**Reply by zer0uno** [reply]
@mhils

> If you want the proxy to just pass through data without looking into it

That's exactly what I would like to achieve just for experimentation purposes. Is it possible with mitmproxy?

**Reply by mhils** [reply]
Yes, see https://docs.mitmproxy.org/stable/howto-ignoredomains/. :)

**Reply by zer0uno** [reply]
thanks, that's fabulous!
